### PR TITLE
diff-so-fancy: Add version 1.4.3

### PR DESCRIPTION
**Pull request by MartinDelille**
It doesn't work yet but I'll be happy to have some help because this is my first formula.

I currently have the following error:

```
> diff-so-fancy
The system cannot find the path specified.
```

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
Refer to this manifest, it's extremely similar - https://github.com/ScoopInstaller/Main/blob/master/bucket/neofetch.json

**Comment by MartinDelille** [comment]
I updated the manifest but still have the issue. In order to make it work from the cloned repository I have to type 'perl /path/to/diff-so-fancy/diff-so-fancy'

**Comment by rashil2000** [comment]
Use this:
```json
{
    "version": "1.4.3",
    "description": "Make diffs human readable instead of machine readable to improve code quality and spot defects faster.",
    "homepage": "https://github.com/so-fancy/diff-so-fancy",
    "license": "MIT",
    "suggest": {
        "git": "git"
    },
    "url": "https://github.com/so-fancy/diff-so-fancy/releases/download/v1.4.3/diff-so-fancy#/diff-so-fancy.pl",
    "hash": "de90659705436cf17d9a8aaad7012c9ed82d7f16eadf1ecee3d6badb0717b73f",
    "pre_install": [
        "@(",
        "   \"#!/bin/sh\",",
        "   \"`\"$dir\\diff-so-fancy.pl`\" `\"$@`\"\"",
        ") -join \"`n\" | Out-File \"$dir\\diff-so-fancy\" -Encoding ASCII -NoNewline"
    ],
    "bin": "diff-so-fancy",
    "checkver": "github",
    "autoupdate": {
        "url": "https://github.com/so-fancy/diff-so-fancy/releases/download/v$version/diff-so-fancy#/diff-so-fancy.pl"
    }
}
```

**Comment by MartinDelille** [comment]
Ok it works now! You gave me the clue with the `pre_install` step but I used the [ack manifest](https://github.com/ScoopInstaller/Main/blob/master/bucket/ack.json) which is the same kind of perl script.

File: bucket/diff-so-fancy.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "version": "1.4.3",
+    "description": "Make diffs human readable instead of machine readable to improve code quality and spot defects faster.",
+    "homepage": "https://github.com/so-fancy/diff-so-fancy",
+    "license": "MIT",
+    "suggest": {
+        "git": "git"
```
Comment (rashil2000) [inline bucket/diff-so-fancy.json:7]:
```suggestion
        "Perl": "perl"
```

File: bucket/diff-so-fancy.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    "url": "https://github.com/so-fancy/diff-so-fancy/releases/download/v1.4.3/diff-so-fancy#/diff-so-fancy",
```
Comment (rashil2000) [inline bucket/diff-so-fancy.json:9]:
```suggestion
    "url": "https://github.com/so-fancy/diff-so-fancy/releases/download/v1.4.3/diff-so-fancy",
```

File: bucket/diff-so-fancy.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "hash": "de90659705436cf17d9a8aaad7012c9ed82d7f16eadf1ecee3d6badb0717b73f",
+    "pre_install": "Set-Content -Value '@perl.exe \"%~dp0diff-so-fancy\" %*' -Path \"$dir\\diff-so-fancy.bat\"",
+    "bin": "diff-so-fancy.bat",
+    "checkver": "github",
+    "autoupdate": {
+        "url": "https://github.com/so-fancy/diff-so-fancy/releases/download/v$version/diff-so-fancy#/diff-so-fancy"
```
Comment (rashil2000) [inline bucket/diff-so-fancy.json:15]:
```suggestion
        "url": "https://github.com/so-fancy/diff-so-fancy/releases/download/v$version/diff-so-fancy"
```

**Pull request merged**